pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -386,6 +386,49 @@ type ObjectStoreSpec struct {
 
 	// The rgw pod info
 	Gateway GatewaySpec `json:"gateway"`
+
+	// The multisite info
+	Multisite MultisiteSpec `json:"multisite"`
+}
+
+type GatewaySpec struct {
+	// The port the rgw service will be listening on (http)
+	Port int32 `json:"port"`
+
+	// The port the rgw service will be listening on (https)
+	SecurePort int32 `json:"securePort"`
+
+	// The number of pods in the rgw replicaset. If "allNodes" is specified, a daemonset is created.
+	Instances int32 `json:"instances"`
+
+	// Whether the rgw pods should be started as a daemonset on all nodes
+	AllNodes bool `json:"allNodes"`
+
+	// The name of the secret that stores the ssl certificate for secure rgw connections
+	SSLCertificateRef string `json:"sslCertificateRef"`
+
+	// The affinity to place the rgw pods (default is to place on any available node)
+	Placement rookv1.Placement `json:"placement"`
+
+	// The annotations-related configuration to add/set on each Pod related object.
+	Annotations rookv1.Annotations `json:"annotations,omitempty"`
+
+	// The resource requirements for the rgw pods
+	Resources v1.ResourceRequirements `json:"resources"`
+
+	// PriorityClassName sets priority classes on the rgw pods
+	PriorityClassName string `json:"priorityClassName,omitempty"`
+}
+
+type MultisiteSpec struct {
```

Comment:
Wait, why does the multisite spec need all three of these properties? The Zone CR points to its ZoneGroup, and the ZoneGroup points to its realm, right? So it seems like we just need to set the zone here, and everything else we can get from the other CRs.